refinery/tool_manager/tests.py

```diff
@@ -1361,6 +1375,302 @@ def test_galaxy_history_id(self):
+        self.create_valid_tool(ToolDefinition.WORKFLOW)
+        self.tool.update_galaxy_data(
+            self.tool.GALAXY_WORKFLOW_INVOCATION_DATA,
+            {
+                "id": self.GALAXY_ID_MOCK
+            }
+        )
+
+        task_list = _get_galaxy_download_tasks(self.tool.analysis)
+        self.assertTrue(show_matching_datasets_mock.called)
+        self.assertTrue(show_invocation_mock.called)
+        self.assertTrue(get_history_file_list_mock.called)
+
+        self.assertEqual(AnalysisResult.objects.count(), 3)
+
+        # There will be one less WorkflowFilesDL because one of our mock
+        # datasets has been "purged"
+        self.assertEqual(WorkflowFilesDL.objects.count(), 2)
+        self.assertEqual(
+            AnalysisResult.objects.count(),
+            self.tool.analysis.results.all().count()
+        )
+        self.assertEqual(len(task_list), 3)
+
+    @mock.patch(
+        "bioblend.galaxy.workflows.WorkflowClient.show_invocation",
+        return_value=galaxy_workflow_invocation
+    )
+    @mock.patch(
+        "bioblend.galaxy.histories.HistoryClient.show_matching_datasets",
+        return_value=galaxy_datasets_list
+    )
+    def test_create_analysis_node_connection_outputs(
+            self,
+            show_matching_datasets_mock,
+            show_invocation_mock
+    ):
+        self.create_valid_tool(ToolDefinition.WORKFLOW)
+        self.tool.update_galaxy_data(
+            self.tool.GALAXY_WORKFLOW_INVOCATION_DATA,
+            {
+                "id": self.GALAXY_ID_MOCK
+            }
+        )
+        self.tool.create_analysis_node_connections(input_nodes=False)
+        self.assertEqual(AnalysisNodeConnection.objects.count(), 3)
+        self.assertEqual(
+            AnalysisNodeConnection.objects.filter(
+                direction=INPUT_CONNECTION
+            ).count(),
+            1
+        )
+        self.assertEqual(
+            AnalysisNodeConnection.objects.filter(
+                direction=OUTPUT_CONNECTION
+            ).count(),
+            2
+        )
+
+        self.assertTrue(show_matching_datasets_mock.called)
+        self.assertTrue(show_invocation_mock.called)
+
+    def test_creating__workflow_tool_sets_tool_launch_config_galaxy_data(self):
+        self.create_valid_tool(ToolDefinition.WORKFLOW)
+        self.update_nodes()
+        self.assertEqual(
+            self.tool.get_galaxy_dict()[
+                WorkflowTool.FILE_RELATIONSHIPS_GALAXY
+            ],
+            unicode(
+                self.tool.get_galaxy_file_relationships()
+            ).replace("'", '"')
+        )
+        self.assertEqual(
+            self.tool.get_galaxy_dict()[
+                WorkflowTool.GALAXY_TO_REFINERY_MAPPING_LIST],
+            []
+        )
+
+    @mock.patch.object(WorkflowTool, "_get_tool_data")
+    def test_get_tool_launch_config(self, get_tool_data_mock):
+        self.create_valid_tool(ToolDefinition.WORKFLOW)
```

Comment:
I think `self.create_valid_tool(ToolDefinition.WORKFLOW)` could be done in setUp() and save a lot of repetition.